### PR TITLE
Move definitions pull from config.yml to separate script

### PR DESCRIPTION
**Pull request by RTann**
Many ideas are from https://github.com/stackrox/stackrox

### CODE REVIEW

File: scripts/ci/lib.sh [hunk 1]
```diff
@@ -1,11 +1,48 @@
...
+        [ -n "${CIRCLE_PROJECT_USERNAME}" ] || { echo "CIRCLE_PROJECT_USERNAME not found" ; exit 2; }
+        [ -n "${CIRCLE_PROJECT_REPONAME}" ] || { echo "CIRCLE_PROJECT_REPONAME not found" ; exit 2; }
+        pull_request="${CIRCLE_PULL_REQUEST}"
+        org="${CIRCLE_PROJECT_USERNAME}"
+        repo="${CIRCLE_PROJECT_REPONAME}"
+    else
+        die "not supported"
+    fi
+
+    headers=()
```
Comment (RTann) [inline scripts/ci/lib.sh:29]: should this be `local`, too? I just copied it from https://github.com/stackrox/stackrox/blob/master/scripts/ci/lib.sh#L422

File: scripts/ci/lib.sh [hunk 2]
```diff
@@ -1,11 +1,48 @@
...
+    if [[ -n "${GITHUB_TOKEN:-}" ]]; then
+        headers+=(-H "Authorization: token ${GITHUB_TOKEN}")
+    fi
+
+    url="https://api.github.com/repos/${org}/${repo}/pulls/${pull_request}"
```
Comment (RTann) [inline scripts/ci/lib.sh:34]: same with this

**Review by gavin-stackrox — APPROVED** [review]

**Comment by RTann** [comment]
hmm I noticed `jq: error (at <stdin>:4): Cannot iterate over null (null)` in the output. Going to look into this before merging

File: scripts/ci/lib.sh [hunk 3]
```diff
@@ -1,11 +1,48 @@
...
+    if [[ -n "${GITHUB_TOKEN:-}" ]]; then
+        headers+=(-H "Authorization: token ${GITHUB_TOKEN}")
+    fi
+
+    local url="https://api.github.com/repos/${org}/${repo}/pulls/${pull_request}"
+    curl -sS "${headers[@]}" "${url}"
+}
+
+pr_has_label() {
+    if [[ -z "${1:-}" ]]; then
```
Comment (jvdm) [inline scripts/ci/lib.sh:39]:
```suggestion
    if [[ $# -lt 1 ]]; then
```

File: scripts/lib.sh [hunk 4]
```diff
@@ -58,7 +58,7 @@ require_environment() {
             fi
             die "$message"
         fi
-    )
+    ) || exit 1
```
Comment (jvdm) [inline scripts/lib.sh:61]: Why are we running this piece of the script in a sub-shell?

File: scripts/ci/lib.sh [hunk 5]
```diff
@@ -31,7 +68,7 @@ push_images() {
         local tag="$2"
 
         for image in "${image_set[@]}"; do
-            "$ROOT/scripts/push-as-manifest-list.sh" "${registry}/${image}:${tag}" | cat
+            "$SCRIPTS_ROOT/scripts/push-as-manifest-list.sh" "${registry}/${image}:${tag}" | cat
```
Comment (jvdm) [inline scripts/ci/lib.sh:71]: This pipe to cat seems redundant.

Comment (jvdm) [reply scripts/ci/lib.sh:71]: I see; The pipe enforces stdout to not appear as a TTY for whatever is running inside `push-as-manifestt-Iist.sh`. Thanks.

**Comment by RTann** [comment]
Closing this in favor of waiting until the stackrox repo has migrated some more so we can copy and paste

**Pull request closed without merging**